src/lib/syscall-logger/src/lib.rs

```diff
@@ -60,7 +60,7 @@ pub fn log_syscall(args: TokenStream, input: TokenStream) -> TokenStream {
     let context_arg_name;
 
     {
-        let mut input_fn = match &mut input {
+        let input_fn = match &mut input {
```

Comment:
At first glance it's not obvious how this compiles without the `mut`. Is it because  the inner captured type is already a `&mut`? Maybe it'd be helpful to have an explicit type in the `let`, if it's not horribly verbose.

Reply:
Yeah, since just the reference has to be mutable, but the thing holding the reference (`input_fn`) doesn't need to be mutable. And if you match on a reference to an enum, the associated data for the variant will also be a reference.

I think it's similar to the following:

```rust
let mut x = Some(1);
let y = match &mut x {
    Some(a) => a,
    None => panic!(),
};
*y = 5;
```

Reply:
I added a `&mut _` to show that it's a reference.

Reply:
Thanks, LGTM!